virttest/utils_test/__init__.py

```diff
@@ -1765,36 +1765,45 @@ def __init__(self, vm, stress_type):
 
         def _parameters_filter(stress_type):
             """Set parameters according stress_type"""
-            _control_files = {'unixbench': "unixbench5.control",
-                              'stress': "stress.control",
-                              'iozone': "iozone.control"}
-            _check_cmds = {'unixbench': "pidof -s ./Run",
-                           'stress': "pidof -s stress",
-                           'iozone': "pidof -s iozone"}
-            _stop_cmds = {'unixbench': "killall ./Run",
-                          'stress': "killall stress",
-                          'iozone': "killall iozone"}
+            _download_url = {
+                'stress': 'http://people.seas.harvard.edu/~apw/stress/stress-1.0.4.tar.gz',
+                'iozone': 'http://www.iozone.org/src/stable/iozone3_458.tar'}
+            _tarball_name = {'stress': 'stress-1.0.4.tar.gz',
+                             'iozone': 'iozone3_458.tar'}
+            _pkg_md5 = {'stress': '890a4236dd1656792f3ef9a190cf99ef',
+                        'iozone': 'de170c3ae60d48ebc331d16beaaa01cb'}
+            _make_cmds = {'stress': './configure && make && make install',
+                          'iozone': 'cd src/current && make linux'}
```

Comment:
I was thinking something like this,
```python
class VMStress(object):
    def __init__(self, vm, stress_type):
        self.download_url = self.params.get("%s_url" % stress_type, "")
        self.tarball_name = self.params.get("%s_tarball" % stress_type, "")
```
where the params are defined in cfg,
```
# params for stress tests
stress_args = "stress --io 5 --vm 5 --cpu 5"
stress_url = "http://people.seas.harvard.edu/~apw/stress/stress-1.0.4.tar.gz"
stress_tarball = "stress-1.0.4.tar.gz"
download_type = "wget"
check_cmd = "example"

# params for iozone tests
iozone_url = "http://www.iozone.org/src/stable/iozone3_458.tar"
iozone_tarball = "iozone3_458.tar"
download_type = "wget"
check_cmd = "example"
```
```python
self.download(download_type):
    """
    :param type: git, wget to be used to download the package
    """
self.install(make_cmd)
self.run()
self.check(check_cmd)
self.clean()
```
This should help adding new one in future by simple configuration changes and minimal code changes respective to that new test,
new_args = "new --option1 --option2"
new_test_object = VMStress(vm, new)

Reply:
@balamuruhans Yes, that's better for extension, I will refine it. Thanks.

Reply:
@balamuruhans I still prefer give a series default value. Since it seems not proper to put these values into somewhere like base.cfg, so if we define them in each specific test case cfg, we will have many same parameters everywhere in different cfg. These values are not like qemu test matrix parameters, they are not likely to change frequently for different test cases.
I agree that it might prevent the convenience for future extension. So I might implement it with @aliang123 's option, allowing pass default values when initialize the class, as well as define them in cfg. Something like:
```
class VMStress(object):
    def __init__(self, vm, stress_type, download_url="", tarball_name=""):
        self.download_url = self.params.get("%s_url" % stress_type, download_url)
        self.tarball_name = self.params.get("%s_tarball" % stress_type, tarball_name)
```

Reply:
@vivianQizhu @aliang123 I agree with you guys.

Reply:
@vivianQizhu This PR will be good a feature for multiple tests, Thanks for it!